mobile/src/main/java/ch/epfl/sdp/mobile/application/tournaments/TournamentFacade.kt

```diff
@@ -23,9 +25,34 @@ class TournamentFacade(private val auth: Auth, private val store: Store) {
    * Allows a user to create a tournament. The user in question administrates the tournament.
    *
    * @param user The [AuthenticatedUser] that wants to join the "Tournament".
-   * @param parameters The "TournamentParameters" that parametrize the user's "Tournament".
+   * @param name The name of the tournament.
+   * @param maxPlayers The maximal number of players allowed to join the tournament.
+   * @param bestOf The number of best-of rounds.
+   * @param poolSize The target size of each pool.
+   * @param eliminationRounds The number of direct elimination rounds. Directly influences number of
+   * player selected from the pool phase.
    */
-  // TODO: Add the function corresponding to the documentation right above
+  suspend fun createTournament(
+      user: AuthenticatedUser,
+      name: String,
+      maxPlayers: Int,
+      bestOf: Int,
+      poolSize: Int,
+      eliminationRounds: Int,
+  ): TournamentReference {
+    val document = store.collection("tournaments").document()
+    document.set(
+        TournamentDocument(
+            adminId = user.uid,
+            name = name,
+            maxPlayers = maxPlayers,
+            bestOf = bestOf,
+            poolSize = poolSize,
+            eliminationRounds = eliminationRounds,
+        ))
```

Comment:
You should validate the parameters here, since the UI might input inconsistent data.

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/336/commits/91e51180395a60d4211803e7d03d421ef259cd5f

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/336/commits/7e074659784f50296983c768153080ee3d2f20e6, for real this time!